### PR TITLE
lab/toolchain-decompilare: Added solutions, Modified README

### PR DESCRIPTION
**Pull request by RaresCon**

### CODE REVIEW

File: laborator/content/toolchain-decompilare/2-warm-up-gotos/Makefile
Comment (teodutu) [inline laborator/content/toolchain-decompilare/2-warm-up-gotos/Makefile:1]: Why did you remove these files?

File: laborator/content/toolchain-decompilare/README.md [hunk 1]
```diff
@@ -255,11 +255,11 @@ Pentru algoritmii de mai jos scrieți cod în C fără a folosi:
 
 Singura instrucțiune permisă în cadrul unui if este `goto`.
 
-În alte cuvinte, tot codul trebuie să fie scris în interiorul funcției main, iar modificarea fluxului de control (saltul la altă zonă de cod) se face doar prin intermediul secvențelor de tipul if (conditie) goto eticheta; sau goto eticheta;.
+În alte cuvinte, tot codul trebuie să fie scris în interiorul funcției main, iar modificarea fluxului de control (saltul la altă zonă de cod) se face doar prin intermediul secvențelor de tipul if (conditie) **goto** eticheta; sau **goto** eticheta;.
```
Comment (teodutu) [inline laborator/content/toolchain-decompilare/README.md:258]:
```suggestion
În alte cuvinte, tot codul trebuie să fie scris în interiorul funcției main, iar modificarea fluxului de control (saltul la altă zonă de cod) se face doar prin intermediul secvențelor de tipul `if (conditie) goto eticheta;` sau `goto eticheta;`.
```
Use backquotes for (pseudo-)code snippets.

File: laborator/content/toolchain-decompilare/README.md [hunk 2]
```diff
@@ -225,9 +225,9 @@ Observați codul în limbaj de asamblare atunci când opțiunile de compilare (`
 
 Intrați în directorul `2-warm-up-gotos`.<br>
 
-**2.1** Modificați codul sursă din fișierul `bogosort.c` ([Bogosort](https://en.wikipedia.org/wiki/Bogosort)) prin înlocuirea instrucțiunii break cu o instrucțiune goto astfel încât funcționalitatea să se păstreze.
+**2.1** Modificați codul sursă din fișierul `bogosort.c`, aflat în directorul `a-bogosort`, ([Bogosort](https://en.wikipedia.org/wiki/Bogosort)) prin înlocuirea instrucțiunii **break** cu o instrucțiune **goto** astfel încât funcționalitatea să se păstreze.
 
-**2.2** În mod asemănător modificați instrucțiunea continue din ignore_the_comments.c astfel încât funcționalitatea codului să se păstreze.
+**2.2** În mod asemănător modificați instrucțiunea **continue** din `ignore_the_comments.c`, aflat în directorul `b-ignore_the_comments`, astfel încât funcționalitatea codului să se păstreze.
```
Comment (teodutu) [inline laborator/content/toolchain-decompilare/README.md:230]:
```suggestion
**2.2** În mod asemănător modificați instrucțiunea `continue` din `ignore_the_comments.c`, aflat în directorul `b-ignore_the_comments`, astfel încât funcționalitatea codului să se păstreze.
```

File: laborator/content/toolchain-decompilare/README.md [hunk 3]
```diff
@@ -225,9 +225,9 @@ Observați codul în limbaj de asamblare atunci când opțiunile de compilare (`
 
 Intrați în directorul `2-warm-up-gotos`.<br>
 
-**2.1** Modificați codul sursă din fișierul `bogosort.c` ([Bogosort](https://en.wikipedia.org/wiki/Bogosort)) prin înlocuirea instrucțiunii break cu o instrucțiune goto astfel încât funcționalitatea să se păstreze.
+**2.1** Modificați codul sursă din fișierul `bogosort.c`, aflat în directorul `a-bogosort`, ([Bogosort](https://en.wikipedia.org/wiki/Bogosort)) prin înlocuirea instrucțiunii **break** cu o instrucțiune **goto** astfel încât funcționalitatea să se păstreze.
```
Comment (teodutu) [inline laborator/content/toolchain-decompilare/README.md:228]:
```suggestion
**2.1** Modificați codul sursă din fișierul `bogosort.c`, aflat în directorul `a-bogosort`, ([Bogosort](https://en.wikipedia.org/wiki/Bogosort)) prin înlocuirea instrucțiunii `break` cu o instrucțiune `goto` astfel încât funcționalitatea să se păstreze.
```

**Review by teodutu — CHANGES_REQUESTED** [review]
You forgot to add the solutions and removed some exercises that you shouldn't have.

**Comment by RaresCon** [comment]
> You forgot to add the solutions and removed some exercises that you shouldn't have.

Don't really know what happened there, as I've made sure to add all changes. I revised the last commit to include the files.

**Review by teodutu — CHANGES_REQUESTED** [review]
I like your changes to exercises 2 and 3. In addition to my comments, mind the `checkpatch.pl` errors: https://github.com/systems-cs-pub-ro/iocla/actions/runs/4518662503/jobs/7958678153?pr=153.

File: laborator/content/toolchain-decompilare/solution/5-old-hits/README [hunk 4]
```diff
@@ -0,0 +1,19 @@
+Exercițiul necesită găsirea unei valori generate random în main folosind gdb
```
[the remark this answers is not in the document]
Comment (RaresCon) [reply laborator/content/toolchain-decompilare/solution/5-old-hits/README:1]: Should I add that title to the README?

**Comment by RaresCon** [comment]
> I like your changes to exercises 2 and 3. In addition to my comments, mind the `checkpatch.pl` errors: https://github.com/systems-cs-pub-ro/iocla/actions/runs/4518662503/jobs/7958678153?pr=153.

I get this error when running `checkpatch.pl` on `old-hits.c`: `Error: WARNING: break quoted strings at a space character`. It is coming from the next declaration:

`char links[] =
		"https://www.youtube.com/watch?v=dQw4w9WgXcQ"
		"https://www.youtube.com/watch?v=3rzgrP7VA_Q"
		"https://www.youtube.com/watch?v=ZZ5LpwO-An4"
		"https://www.youtube.com/watch?v=y6120QOlsfU"
		"https://www.youtube.com/watch?v=0q6yphdZhUA"
		"https://www.youtube.com/watch?v=989-7xsRLR4"
		"https://www.youtube.com/watch?v=7yh9i0PAjck"
		"https://www.youtube.com/watch?v=04F4xlWSFh0"
		"https://www.youtube.com/watch?v=hAAlDoAtV7Y"
		"https://www.youtube.com/watch?v=NAEppFUWLfc";`

Any idea how to rezolve this?

File: laborator/content/toolchain-decompilare/solution/5-old-hits/README (hunk 4 above)
[the remark this answers is not in the document]
Comment (teodutu) [reply laborator/content/toolchain-decompilare/solution/5-old-hits/README:1]: Yes. It's in my suggestion: `# Writeup`.

**Comment by teodutu** [comment]
> > I like your changes to exercises 2 and 3. In addition to my comments, mind the `checkpatch.pl` errors: https://github.com/systems-cs-pub-ro/iocla/actions/runs/4518662503/jobs/7958678153?pr=153.
> 
> I get this error when running `checkpatch.pl` on `old-hits.c`: `Error: WARNING: break quoted strings at a space character`. It is coming from the next declaration:
> 
> `char links[] = "https://www.youtube.com/watch?v=dQw4w9WgXcQ" "https://www.youtube.com/watch?v=3rzgrP7VA_Q" "https://www.youtube.com/watch?v=ZZ5LpwO-An4" "https://www.youtube.com/watch?v=y6120QOlsfU" "https://www.youtube.com/watch?v=0q6yphdZhUA" "https://www.youtube.com/watch?v=989-7xsRLR4" "https://www.youtube.com/watch?v=7yh9i0PAjck" "https://www.youtube.com/watch?v=04F4xlWSFh0" "https://www.youtube.com/watch?v=hAAlDoAtV7Y" "https://www.youtube.com/watch?v=NAEppFUWLfc";`
> 
> Any idea how to rezolve this?

The implementation of `old-hits.c` is kind of messy because it considers all links are one contiguous string. To fix this, change the `links` array to a array of strings and copy them to a contiguous buffer which you'll then encrypt instead of `links`.

**Comment by RaresCon** [comment]
I've fixed the errors given by `checkpatch.pl`, but I see that it still tracks old errors, that are not present in the current commit.

**Comment by teodutu** [comment]
> I've fixed the errors given by `checkpatch.pl`, but I see that it still tracks old errors, that are not present in the current commit.

This is because `checkpatch.pl` verifies all commits in a PR. If you want to silence previous errors, you have to squash your broken commits into a single, correct one.

**Review by teodutu — CHANGES_REQUESTED** [review]
Just make a slight improvement to `old-hits.c` and then it'll be fine.

**Review by teodutu — APPROVED** [review]
:ok_hand:

**Pull request merged**